docs/Internals/bcdb/bcdb.md

```diff
@@ -0,0 +1,157 @@
+one BCDB object corresponds to one database. 
+
+it is identified by its name e.g. test
+you can have multiple schema from one url (different versions)
+
+### Schemas  
+we can access the schemas dictionnary in `bcdb.meta.data.schemas`
+A Model is based on only one schema. think of the schema as a text defining the structure 
+of a data table and the model is the databale object.
+To get a model corresponding to a schema we have several options 
+* through the schema text itself 
+    * the fastest and more straight forward method  `bcdb.model_get_from_schema()`
+* through the property url of the schema  
+    * we find the corresponding schema text based on the schema url to call the method `bcdb.model_get_from_schema()`
+* through the md5 hash of the schema  
+    * we find the corresponding schema text based on the schema md5 hash to call the method `bcdb.model_get_from_schema()`
+* through the schema id 
+    * we find the corresponding schema text based on the schema id to call the method `bcdb.model_get_from_schema()`
+
+### BCDBModel
+A BCDBModel object is iterable and take the namespace id as parameter.
+```
+JSX> model = bcdb.model_get_from_schema(schema_text)
+JSX> model.iterate($nid)
+```
+Or ,as the ID is incremented for all the database, meaning that you __will not have the same object id__ for objects in different namespace, we can get the object directly with its id.
+```
+JSX> model($id)
+```
+
+
+## BCDB Virtual file system
+The idea is to have a VFS as a front to BCDB and adaptaters that will take advantage of this VFS to provide extra connectivity
+![VFS](images/BCDB_VFS.png)
+
+it allows you to navigate through the BCDB like it was a file system.
+the root directory is the bcdb name.
+
+
+Here is the file system directories
+```
+    /data
+        /nid
+            /sid
+                /1
+                    object1
+                    object2
+            /hash
+                /0eccf565df45
+                    object1
+                    object2
+            /url
+                /ben.test.1
+                    object1
+                    object2
+    /schemas
+        /sid
+            1
+        /hash
+            0eccf565df45
+        /url
+            ben.test.1
+        /url2sid
+            ben.test.1
+    info
+```   
+eg. test/data/2/url/ben.test.1/object1, test/schemas/hash/0eccf565df45 
+
+if bcdb name is set eg. /data/2/url/ben.test.1/, data/2/url/ben.test.1/object2,
+
+On each level we can do 
+- get 
+    - retrieve a file or a directory given the specified path
+- len
+    - file length if the current path is a file
+    - throws if the current path is a directory
+- set
+    - set a file inside the current path
+    - throws if the current path is a directory
+- list 
+    - List the files of a directory
+    - throws if the current path is a file
+- delete
+    - delete a file or a directory given the specified path
+
+  
+
+### client to DB
+zdb client communicate with 0db and uses redis protocol
```

Comment:
protocol is called RESP (https://redis.io/topics/protocol)

Reply:
je rajoute cela merci @zaibon